layouts/_default/_markup/render-link.html

```diff
@@ -1,16 +1,7 @@
-{{ $link := .Destination -}}
-{{ $isRemote := strings.HasPrefix $link "http" -}}
-{{ if not $isRemote -}}
-  {{ $url := urls.Parse .Destination -}}
-  {{ if $url.Path -}}
-    {{ $fragment := "" -}}
-    {{ with $url.Fragment }}{{ $fragment = printf "#%s" . }}{{ end -}}
-    {{ with .Page.GetPage $url.Path }}{{ $link = printf "%s%s" .RelPermalink $fragment }}{{ end -}}
-  {{ end -}}
-{{ end -}}
-<a href="{{ $link | safeURL }}"
+{{ $isExternal := hasPrefix .Destination "http" -}}
+<a href="{{ .Destination | safeURL }}"
   {{- with .Title}} title="{{ . }}"{{ end -}}
-  {{- if $isRemote }} target="_blank" rel="noopener"{{ end -}}
+  {{- if $isExternal }} target="_blank" rel="noopener"{{ end -}}
```

Comment:
Btw if, at some point, we'd rather not use CSS to add the external link icon, we could do so programatically here. But for now, IMHO, the solution we have in place is fine.

/cc @kapunahelewong